### DISCUSSION TITLE
Sulcal curvature metrics

### DISCUSSION THREAD

**Discussion by wsnyder4** [brainvisa/brainvisa.github.io#111] [Q&A] [answered]
Dear BrainVISA experts,

I have two questions regarding the sulciCurvature.py script, and was curious if anyone has insight into what the measures mean. Firstly, how do the available methods to choose from (FEM, Barycenter, Boix, Boixgaussian) differ from concepts like mean curvature and gaussian curvature?
Second, why does the script only retain the largest connected component for each sulcus? I notice that the mesher command will only mesh the largest connected component, given by:
``
m = aimsalgo.Mesher()
``

 in addition to AimsConnectComp explicitly retaining the largest connected component in this script:

``soma.subprocess.call(['AimsConnectComp', '-i', tmpvolfile, '-o', tmpvolfile,
                          '-c', '18', '-n', '1', '-b'], stdout=tf[0])
 ``

Thank you!
All the best,
Will

**Comment by denisri** [comment, marked as answer]
Hi Will,
This script is rather old and I don't remember everything about it. I more or less remember I have written it for a specific study (more than 10 years ago) where I was interested in large sulci, and possibly only the largest part of them, so it was "sufficient" for that use by that time. I agree the script misses other connected components and if you are interested in such measurements the script probably needs some reworking.
- The mesh curvature methods have been implemented about 20 years ago (by Arnaud Cachia if I remember, who was a student at that time), I'm not a specialist of curvature calculation but there are several more or less standard ways of estimating / approximating it in more or less robust ways. There are certainly many papers describing this topic (I find for instance https://hal.archives-ouvertes.fr/hal-01024747/): "FEM" is a finite element approximation, "Boix" is (I thing) the name of the author of the method (see the "Boix95" reference in the above paper)...
- Here curvature is estimated on every vertex of a mesh of the sulcus, but the meshes we are using "go around" the skeleton voxels of the sulcus (both sides of the object), thus have a very high and "artificial" curvature on the edges of the meshed object (bottom and external lines of the sulcus especially), so they need to be cleaned to removes those points. I don't remember exactly if this cleaning is good enough...
- The connected component thing is, I think, actually, a technical constraint because the `Mesher.getBrain()` function used to remesh the sulcal object only meshes the first connected component of the object. `Mesher` has other methods which builds meshes for every object, and even every interface of every object (if the object has holes, for instance), but here we have used `getBrain()` for convenience, because this function was performing all meshing, decimation, and slight smoothing steps, which also removed some spurious high curvature points, but this could be done for every component, either by doing the same for each connected component, or using other `Mesher` functions (`Mesher.diot()`, `Mesher.decimate()`, `Mesher.smooth()`...)

I hope this helps a bit...
Denis

**Reply by wsnyder4** [reply]
Thank you for the thorough reply Denis!
The "cleaning" performed seems to work quite while, although for some smaller fragments of shallow sulci there can be a behavior where only 2 vertices on either side of the mesh are labeled with a curvature value. I may play around with the distances from hull junction / bottom buckets for retained vertices in average curvature measurement, but overall was excited to see that it worked well with almost all sulci. 
Appreciate the mesher advice as well, I will have to implement this. I have been exporting sulcal meshes to FreeSurfer and using the cleaning and meshing BrainVISA performs as preprocessing for other forms of curvature calculation. (Notably, Barycenter curvature in BV seems to be somewhat related to mean curvature in FS).

Thanks for the guidance!
Will
